src/styles/main.scss

```diff
@@ -1,3 +1,87 @@
+@function calendar-size($cell) {
+  @return $cell * 7 + 10px;
+}
+
+@function start-day($cell,$index) {
+  @return $cell * ($index - 1) + ($index - 1);
+}
+
+@mixin size($size) {
+  width: $size;
+  height: $size;
+}
+
+$cell-size: 100px;
+$calendar-size: calendar-size($cell-size);
+
 body {
   margin: 0;
 }
+
+* {
+  box-sizing: border-box;
+}
+
+.project {
+
+  &__calendar {
+    margin: auto;
+  }
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+  width: $calendar-size;
+  padding-top: $cell-size;
+
+  $days:
+    "mon" 1,
+    "tue" 2,
+    "wed" 3,
+    "thu" 4,
+    "fri" 5,
+    "sat" 6,
+    "sun" 7;
+
+  @each $day,$index in $days {
+    &--start-day--#{$day} :first-child {
+      margin-left: start-day($cell-size,$index);
+    }
+  }
+
+  @for $i from 28 through 31 {
+    &--month-length--#{$i} :nth-child(n+#{$i+1}) {
+      display: none;
+    }
+  }
+
+  &__cell {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+
+    @include size($cell-size);
+
+    font-family: Arial, Helvetica, sans-serif;
+    font-size: 30px;
+
+    border: 1px solid black;
+    background-color: #eee;
+
+    cursor: pointer;
+    transition: 0.5s;
```

Comment:
For now, transition applies to all properties, which is not good for performance.
Specify the exact properties you want apply transition to
https://stackoverflow.com/questions/8947441/css3-transitions-is-transition-all-slower-than-transition-x
![image](https://user-images.githubusercontent.com/96409563/224644503-9f13edbf-bf60-40c9-86d2-16e114ed1623.png)